contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,564 @@
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: block-volume
+provisioner: no-provisioning 
+parameters:
+  volumeType: block
+```
+* Sample of pre-created volume definition:
+
+```
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+ name: pv-block-volume
+spec:
+ volumeType: block
+ storageClassName: block-volume
+ capacity:
+   storage: 35Gi
+ accessModes:
+   - ReadWriteOnce
+ local:
+    path: /dev/xvdc
+```
+## [FUTURE] UC5: 
+
+DESCRIPTION: StorageClass with dynamically created volumes 
+
+BACKGROUND: The admin wishes to create a storage class that will dynamically create block PVs based on a user's PVC request for volumeType: Block. The admin desires the volumes be created dynamically and deleted when the PV definition is deleted. 
+
+WORKFLOW:
+
+ADMIN:
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: local-fast
+provisioner: kubernetes.io/local-block-ssd
+parameters:
+  volumeType: block
+```
+
+***This has implementation details that have yet to be determined. It is included in this proposal for completeness of design ****
+
+## UC6: 
+
+DESCRIPTION: The developer wishes to request a block device via a Storage Class.
+
+WORKFLOW:
+
+USER: 
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+ name: pvc-local-block
+spec:
+ volumeType: block
+ storageClassName: local-fast
+ accessModes:
+  - ReadWriteOnce
+ resources:
+   requests:
+     storage: 10Gi
+```  
+
+## UC7:
+
+DESCRIPTION: User wishes to install a software defined storage solution  (i.e.GlusterFS/Ceph(bluestore)) on top of raw block device. Using something like a StateFulSet for GlusterFS, for example, the system could provide it with provisioned block storage, and allow the software defined storage solution to consume it as needed. 
+
+BACKGROUND: This provides a way to dynamically create raw block devices to later be consumed by software defined storage.
+
+WORKFLOW:
+
+ADMIN:
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: aws-ebs-raw
+provisioner: kubernetes.io/aws-ebs-raw
+parameters:
+  type: gp2
+  fstype: raw 
+```
+
+* Storage Class Stateful Set
+
+```
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 2
+  template:
+    metadata:
+      labels:
+	app: nginx
+    spec:
+      containers:
+      - name: nginx
+	image: gcr.io/google_containers/nginx-slim:0.8
+	ports:
+	- containerPort: 80
+	  name: web
+	volumeMounts:
+	- name: www
+	  mountPath: /usr/share/nginx/html
+ volumeClaimTemplates:
+  - metadata:
+      name: datadir
+      annotations:
+	volume.beta.kubernetes.io/storage-class: aws-ebs-raw
+    spec:
+      accessModes:
+	- "ReadWriteOnce"
+      resources:
+	requests:
+	  storage: 10Gi
+```
+***SUITABLE FOR: NETWORK ATTACHED BLOCK***
+
+## UC8: 
+
+DESCRIPTION: Admin creates network raw block devices
+
+BACKGROUND: Admin wishes to pre-create Persistent Volumes in GCE as raw block devices
```

Comment:
@smarterclayton if you are looking for RWM device, then any iSCSI and FC LUN can be attached to multiple nodes. Applications need to make sure they don't damage data on the device though - they need to use specialized device formats such as clustered LVM or gfs2.